pages/data.py

```diff
@@ -141,12 +160,36 @@ def update_sub_tab(tab, store_demographics):
         df = df.drop(columns=[col for col in df.columns if col not in columns])
 
         return populate_datatable(df)
-      
+
+
+@callback(
+    Output('data-table', 'hidden_columns'), # Output hidden columns in the table
+    Output('button-clicked', 'children'), #updates button label
+    Input('button-clicked', 'n_clicks'), #number of clicks on the button
+    State('button-clicked', 'children') #State representing the current label of button
+)
+#Controls visibility of columns in trips table  and updates the label of button based on the number of clicks.
+def update_dropdowns_trips(n_clicks, button_label):
+    columns = perm_utils.get_allowed_trip_columns()
+    columns.update(
+        col['label'] for col in perm_utils.get_allowed_named_trip_columns()
+    )
+    if n_clicks % 2 == 0:
+        hidden_col = ["data.duration_seconds", "data.distance_meters","data.distance"]
+        button_label = 'Display columns with raw units'
+    else:
+        hidden_columns = ["data.duration", "data.distance_miles", "data.distance_km", "data.distance"]
+        hidden_col = [col for col in hidden_columns if col in columns]
+        button_label = 'Display columns with humanzied units'
+    #return the list of hidden columns and the updated button label
+    return hidden_col, button_label
+
+
 def populate_datatable(df):
     if not isinstance(df, pd.DataFrame):
         raise PreventUpdate
     return dash_table.DataTable(
-        # id='my-table',
+        id= 'data-table', #common id for all table(users,trips,demographics,trajectories)
```

Comment:
ok, so now I am completely confused. Per this comment: https://github.com/e-mission/op-admin-dashboard/pull/97#discussion_r1477659317 you need to have a table-id for the trips table to hide columns. So how does it work to hardcode the same table_id for all tables?

If you read my comments: https://github.com/e-mission/op-admin-dashboard/pull/97#discussion_r1477182533
I still don't see where you indicate to `update_dropdowns_trips` which table ID it is used for. And my question was 

Finally, my comment:
https://github.com/e-mission/op-admin-dashboard/pull/97#issuecomment-1926290830

> Why do you need the trip_id for tables that don't have hidden columns? Can't those invocations remain changed?

is unchanged.

You used to have all the `trip_id`s be `datatable`. They are now `data-table`. Why is this better? And to return to my fundamental question, do you need to have a `table_id` for the tables that don't have hiddent columns?

Reply:
It will not make any difference, most of the id's  provided for tab and buttons are in that format so I just changed it. 

We call same function ```populate_datatable``` to create all the datatables, so either we should provide id for all the table or none of them.

Reply:
@achasmita You have not answered the first part of my question

> ok, so now I am completely confused. Per this comment: https://github.com/e-mission/op-admin-dashboard/pull/97#discussion_r1477659317 you need to have a table-id for the trips table to hide columns. So how does it work to hardcode the same table_id for all tables?

Please read the question carefully and answer comprehensively

Reply:
> Sorry for confusion, there were some extra line of code from previous implementation while using dropdown previously. I have cleaned it now.

<details>

<summary>  

Here I have summarized the changes in trips table:  

</summary>

The ```update_dropdowns_trips``` callback function is responsible for updating the hidden column based on user interaction with the ```Display column with raw units``` button
  
```bash
  @callback(
      Output('data-table', 'hidden_columns'), 
      Output('button-clicked', 'children'), 
      Input('button-clicked', 'n_clicks'), 
      State('button-clicked', 'children') 
  )
  def update_dropdowns_trips(n_clicks, button_label):
      if n_clicks % 2 == 0:
          hidden_col = ["data.duration_seconds", "data.distance_meters","data.distance"]
          button_label = 'Display columns with raw units'
      else:
          hidden_columns = ["data.duration", "data.distance_miles", "data.distance_km", "data.distance"]
          button_label = 'Display columns with humanzied units'
      return hidden_col, button_label
  ```
  It consists of two outputs:
  - hidden_colums
    - Updates the list of hidden columns in the trips table.
    - Controls which columns are visible to the user
  I got the idea of hidden_columns from: https://dash.plotly.com/datatable/interactivity
  
  - button-clicked
    - Based on number of clicks it update the label of the butoon to either ```Display column with raw units``` or  ```Display column with humanized units```
  
  > We need id for table in this callback to specify where the change need to be updated(hidden_columns).
  > We have single function to poputate datatable ```def populate_datatable(df)``` which we is used to populate all datatable and it has single parameter ```df```.
  -  To pass ```id``` to table either I have to use  arguments of dash_table.DataTable ```id``` 
      - where i can directly provide id or pass it as a  new parameter from  ```def populate_datatable(df, table_id)```
  
  Here, I am using common id because the  ```update_dropdowns_trips``` is triggered only when we are inside trip table and interact with button so it will not affect any other table. If it still looks confusing, we can assign unique id to each table.
  
</details>

Reply:
@achasmita I am not asking you to change the code. My question is this:
- before this, the tables did not have ids (at least from what I see in the code)
- you added ids to support hidden columns
- but now you say that `update_dropdowns_trips` is triggered only when we are inside trip table and will not affect any other table
- so why do you need the table id now? Why not just remove it and go back to the code without ids?

Reply:
For ```update_dropdowns_trips``` callback  ID is essential because it specifies the target output where the hidden columns will be updated. Without the ID, it would not know where to apply changes triggered by the callback.

Reply:
@achasmita but with the current code, we have the same callback ID for all tables. So it still doesn't know which table to apply changes triggered by the callback.

Reply:
It will pass the hidden_columns property into populate_datatable only when we are in trips-table tab as the button is inside the trips table. To make it more clear I also have updated code  to pass table id only for trips table.